bids/layout/layout.py

```diff
@@ -550,64 +460,17 @@ def add_derivatives(self, path, parent_database_path=None, **kwargs):
-        def check_for_description(bids_dir):
-            dd = os.path.join(bids_dir, 'dataset_description.json')
-            return os.path.exists(dd)
-
-        for p in paths:
-            p = os.path.abspath(p)
-            if os.path.exists(p):
-                if check_for_description(p):
-                    deriv_dirs.append(p)
-                else:
-                    subdirs = [d for d in os.listdir(p)
-                               if os.path.isdir(os.path.join(p, d))]
-                    for sd in subdirs:
-                        sd = os.path.join(p, sd)
-                        if check_for_description(sd):
-                            deriv_dirs.append(sd)
-
-        if not deriv_dirs:
-
-            warnings.warn("Derivative indexing was requested, but no valid "
-                          "datasets were found in the specified locations "
-                          "({}). Note that all BIDS-Derivatives datasets must"
-                          " meet all the requirements for BIDS-Raw datasets "
-                          "(a common problem is to fail to include a "
-                          "'dataset_description.json' file in derivatives "
-                          "datasets).\n".format(paths) +
-                          "Example contents of 'dataset_description.json':\n%s" %
-                          json.dumps(EXAMPLE_DERIVATIVES_DESCRIPTION))
-        for deriv in deriv_dirs:
-            dd = os.path.join(deriv, 'dataset_description.json')
-            with open(dd, 'r', encoding='utf-8') as ddfd:
-                description = json.load(ddfd)
-            pipeline_name = description.get(
-                'PipelineDescription', {}).get('Name')
-            if pipeline_name is None:
-                raise BIDSDerivativesValidationError(
-                                 "Every valid BIDS-derivatives dataset must "
-                                 "have a PipelineDescription.Name field set "
-                                 "inside 'dataset_description.json'. "
-                                 "\nExample: %s" %
-                                 MANDATORY_DERIVATIVES_FIELDS['PipelineDescription.Name'])
-            if pipeline_name in self.derivatives:
-                raise BIDSDerivativesValidationError(
-                                 "Pipeline name '%s' has already been added "
-                                 "to this BIDSLayout. Every added pipeline "
-                                 "must have a unique name!")
-            # Default config and sources values
-            kwargs['config'] = kwargs.get('config') or ['bids', 'derivatives']
-            kwargs['sources'] = kwargs.get('sources') or self
+        deriv_paths = validate_derivative_paths(paths, self, **kwargs)
+
+        # Default config and sources values
+        kwargs['config'] = kwargs.get('config') or ['bids', 'derivatives']
+        kwargs['sources'] = kwargs.get('sources') or self
```

Comment:
I assume this is only meant to override missing values, not all values that evaluate to `False`?

```suggestion
        kwargs.setdefault('config', ['bids', 'derivatives'])
        kwargs.setdefault('sources', self)
```

Reply:
It's correct this way—this is inside `add_derivatives`, so if we don't replace `None`, the values are set to incorrect defaults downstream.